penman/graph.py

```diff
@@ -173,6 +173,20 @@ def __isub__(self, other):
         else:
             return NotImplemented
 
+    def __contains__(self, item: Union[str, Triple, Tuple[str, str, str]]):
+        """
+        Containment checking of either a string or a triple (tuple). In the case of a
+        string, will return True if the given string is a terminal instance in the graph.
+        If a triple or tuple, will return True if the triple exists in self.triples.
+        """
+        if isinstance(item, str):
+            terminals = [t[2] for t in self.instances()]
+            return item in terminals
+        elif isinstance(item, tuple):
+            return item in self.triples
+        else:
+            raise NotImplemented
```

Comment:
There are a few things to be fixed here.

* Constants can be more than just `str`:
  https://github.com/goodmami/penman/blob/f3b0c423a60f82b13fffeec73fa1a77bf75cd4dc/penman/types.py#L10
* It would be more efficient to assemble `terminals` as a set so the lookup is O(1), and using the `target` accessor instead of `t[2]` is more readable
* The exception class is `NotImplementedError`, and `NotImplemented` is a return value for things like comparisons
* We might not need to raise an exception if we assume non-graphy things are potentially terminals

```suggestion
        if isinstance(item, tuple):
            return item in self.triples
        terminals = {t[2] for t in self.instances()}
        return item in terminals
```